### PR TITLE
[client/rest]: add mosaic redefinition fork handling to rosetta implementation

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: prior to the mosaic redefinition fork, mosaic description changes are considered significant
solution: change comparison of mosaic definitions prior to fork

### CODE REVIEW

File: client/rest/src/plugins/rosetta/nem/OperationParser.js [hunk 1]
```diff
@@ -432,7 +433,8 @@ export class OperationParser {
 
 				const isExistingPurged = transactionLocation && transactionLocation.height > existingMosaicDefinitionTuple.expirationHeight;
 				if (!isExistingPurged) {
-					if (areMosaicDefinitionsEqual(mosaicDefinition, existingMosaicDefinition)) {
+					const isDescriptionSignificant = isMosaicDefinitionDescriptionSignificant(this.network, transactionLocation);
```
Comment (Wayonb) [inline client/rest/src/plugins/rosetta/nem/OperationParser.js:436]: If ``isDescriptionSignificant`` is true, then if the owner is the same, we want to do nothing.
From what I noticed, before the fork, the expired endpoint was triggered when the mosaic was redefined.

File: client/rest/src/plugins/rosetta/nem/OperationParser.js
Comment (Wayonb) [inline client/rest/src/plugins/rosetta/nem/OperationParser.js:1]: In the ``parseBlock``

1. At line 493, the height should be height - 1. In some cases, before the fork, a mosaic redefinition will trigger the expired endpoint for the old definition of the mosaic. So, the new mosaic definition and the expired old definition will be the same height.

```
currency: (await this.options.lookupCurrency(expiredMosaic.mosaicId, { height })).currency,
```

2. at line 515, also check for multi-sig transaction(otherTrans) for mosaic def.
```
const transaction = block.block.transactions[i];
```

**Review by Wayonb — CHANGES_REQUESTED** [review]

File: client/rest/src/plugins/rosetta/nem/rosettaUtils.js [hunk 2]
```diff
@@ -131,3 +132,20 @@ export const areMosaicDefinitionsEqual = (lhs, rhs, isDescriptionSignificant) =>
...
+ * @param {Network} network Blockchain network.
+ * @param {object} transactionLocation Location of transaction.
+ * @returns {boolean} true if mosaic definition description differences are significant.
+ */
+export const isMosaicDefinitionDescriptionSignificant = (network, transactionLocation) => {
+	const forkHeight = (() => {
+		if ('mainnet' === network.name)
+			return 1_110_000;
+
+		return 'testnet' === network.name ? 871_500 : 1;
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply client/rest/src/plugins/rosetta/nem/rosettaUtils.js:147]: would change it later as this is consistent with code in NIS (which means we might need NIS change for testnet) 😅 
see: https://github.com/NemProject/nem/blob/dev/nis/src/main/java/org/nem/nis/BlockMarkerConstants.java#L8

File: client/rest/src/plugins/rosetta/nem/OperationParser.js (hunk 1 above)
Comment (Jaguar0625) [reply client/rest/src/plugins/rosetta/nem/OperationParser.js:436]: so, you're saying we should always treat description as insignificant for purposes of rosetta?

File: client/rest/src/plugins/rosetta/nem/OperationParser.js
Comment (Jaguar0625) [reply client/rest/src/plugins/rosetta/nem/OperationParser.js:1]: were these only changes needed to get Symbol rosetta to work or were there others as well?

Comment (Wayonb) [reply client/rest/src/plugins/rosetta/nem/OperationParser.js:1]: yea, that's all I think

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**